### PR TITLE
feat(select): add `modal` as interface

### PR DESCRIPTION
**Pull request by tanner-reits**
Issue number: resolves internal

---------

## What is the current behavior?

Select only offers `alert`, `action-sheet`, and `popover` as interfaces

## What is the new behavior?

Adds `modal` as an interface option for `ion-select`

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/select-modal/select-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  private findOptionFromEvent(ev: CheckboxCustomEvent | RadioGroupCustomEvent) {
+    const { options } = this;
+    return options.find((o) => o.value === ev.target.value);
+  }
+
+  private getValues(ev?: CheckboxCustomEvent | RadioGroupCustomEvent): string | string[] | undefined {
+    const { multiple, options } = this;
+
+    if (multiple) {
+      // this is a popover with checkboxes (multiple value select)
```
Comment (brandyscarney) [inline core/src/components/select-modal/select-modal.tsx:47]:
```suggestion
      // this is a modal with checkboxes (multiple value select)
```

File: core/src/components/select-modal/select-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      // return an array of all the checked values
+      return options.filter((o) => o.checked).map((o) => o.value);
+    }
+
+    // this is a popover with radio buttons (single value select)
```
Comment (brandyscarney) [inline core/src/components/select-modal/select-modal.tsx:52]:
```suggestion
    // this is a modal with radio buttons (single value select)
```

File: core/src/components/select-modal/select-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    if (option?.handler) {
+      safeCall(option.handler, values);
+    }
+  }
+
+  private setChecked(ev: CheckboxCustomEvent): void {
+    const { multiple } = this;
+    const option = this.findOptionFromEvent(ev);
+
+    // this is a popover with checkboxes (multiple value select)
```
Comment (brandyscarney) [inline core/src/components/select-modal/select-modal.tsx:70]:
```suggestion
    // this is a modal with checkboxes (multiple value select)
```

**Review by brandyscarney — CHANGES_REQUESTED** [review]
This looks good overall, I left some feedback on the JIRA ticket though.

**Review by brandyscarney — APPROVED** [review]
Looks good!

**Comment by Asko-Dev** [comment]
Hi. Love the modal option ... didn't find though how to change the `Close` text on the close button without resorting to passing in entire component in options? Is it possible somehow? My app is in czech.

**Pull request merged**